lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -62,23 +62,31 @@ def get_paginated_obj(self, query: Query, json: JSON, url: str, page_limit: int)
         """
         paginated_result = {}
         total_count = query.count()
-        if total_count > page_limit:
-            if "page_no" in json:
-                items = (
-                    query.order_by(Dataset.name)
-                    .limit(page_limit)
-                    .offset((json["page_no"] - 1) * page_limit)
-                    .all()
-                )
-                json["page_no"] += 1
-            else:
-                # This is the first page
-                items = query.limit(page_limit).all()
-                json["page_no"] = 1
-            paginated_result["next_url"] = url.rsplit("?", 1)[0] + "?" + urlencode(json)
-            paginated_result["total"] = total_count
+        page_limit = json.get(
+            "page_limit", int(self.config.server_config.get("page_limit"))
+        )
+        page_no = json.get("page_no", 1)
+        items = (
+            query.order_by(Dataset.name)
+            .limit(page_limit)
+            .offset((page_no - 1) * page_limit)
+            .all()
+        )
+        json["page_no"] = page_no + 1
+        parsed_url = urlparse(url)
+        if len(items) < page_limit:
+            next_url = ""
         else:
-            items = query.order_by(Dataset.name).all()
+            next_url = (
+                parsed_url.scheme
+                + "://"
+                + parsed_url.netloc
+                + parsed_url.path
+                + "?"
+                + urlencode(json)
+            )
+        paginated_result["next_url"] = next_url
```

Comment:
If there is no "next URL" (e.g., because we set it to an empty string), would you rather just omit it from the JSON object?  (That's easy to do....)